src/main/java/command/TaskDoneCommand.java

```diff
@@ -0,0 +1,48 @@
+package com.leeyiyuan.command;
+
+
+import com.leeyiyuan.storage.Storage;
+import com.leeyiyuan.storage.StorageException;
+import com.leeyiyuan.task.Task;
+import com.leeyiyuan.task.TaskList;
+import com.leeyiyuan.ui.UserOutputInterface;
+
+/** 
+ * Represents a Command to mark an existing Task as done. 
+ */
+public class TaskDoneCommand extends Command {
+
+    /** Index of the Task to mark as done. */
+    protected int index;
+
+    /**
+     * Constructs a TaskDoneCommand from an index.
+     *
+     * @param index The index of the Task to mark as done.
+     */
+    public TaskDoneCommand(int index) {
+        this.index = index;
+    }
+
+    /** 
+     * {@inheritDoc} 
+     */
+    @Override
+    public void execute(TaskList tasks, UserOutputInterface uoi, Storage storage)
+            throws CommandExecuteException, StorageException {
+        if (tasks.size() >= this.index) {
+            if (tasks.get(this.index - 1).getIsDone()) {
```

Comment:
Mentioned it in a general comment, but as mentioned, good job covering cases where task is already done :D